### ISSUE TITLE
clusterer.condensed_tree_.plot() pathologically slow for high no. clusters

### ISSUE THREAD

**Issue by thclark** [scikit-learn-contrib/hdbscan#175]
## TL;DR
With a large number of clusters (high dimensional data, wide graph hierarchy), the process hangs whilst plotting the condensed tree.

## Background and use case
I've been using `HDBSCAN` to cluster related strings of text. My test case matrix size`X.shape` is `(8535 x 1820)`. The dataset pops out `~800` clusters, then plotting the condensed tree yields a graph with `~3000` splits.

`hdbscan.plots.CondensedTree.plot()` uses the following code to render the black branch lines:
```python
        for xs, ys in zip(plot_data['line_xs'], plot_data['line_ys']):
            axis.plot(xs, ys, color='black', linewidth=1)
```
However, the looped call to axis.plot() can exhibit pathologically slow behaviour for such high-split cases.

## Reproduction

Recreate the [condensed tree plot](http://hdbscan.readthedocs.io/en/latest/how_hdbscan_works.html#extract-the-clusters) for a dataset where `len(plot_data['line_xs'])` is `~3000` (`plot_data` is the dict returned from calling `clusterer.gen_plot_data()`).

The symptom (may be platform dependent??) is simply a hang of the process (or at least extremely long execution time), due to the extreme number of calls to `axis.plot()`.

## Solution

Pull request inbound. Will be something like:
```
drawlines = []
for xs, ys in zip(plot_data['line_xs'], plot_data['line_ys']):
    drawlines.append(xs)
    drawlines.append(ys)
axis.plot(*drawlines, color='black', linewidth=1)
```
(with thanks to [this SO post](https://stackoverflow.com/questions/43216111/anyway-to-draw-many-lines-fast-in-python-matploblib-is-preferred-if-possible))

~~HELP WANTED: Can someone tell me if that `*` argument unpacking works in python 2.7? I've only ever learned Python3!~~

**Comment by lmcinnes** [comment]
Thanks! I admit I wrote the plotting code fairly quickly and didn't really optimize it much. Any pull requests on this are very greatly appreciated. Definitely looking forward to this!

**Comment by thclark** [comment]
No problem @lmcinnes least I can do after the weeks of hassle this library has saved me. PR just made.

**Comment by thclark** [comment]
Update: Checked that the unpack ("splat") operator works in 2.7.
